app/Dockerfile

```diff
@@ -0,0 +1,63 @@
+# from https://bun.sh/guides/ecosystem/docker, with modifications
+# to run a hot-reloading development server
+
+# use the official Bun image
+# see all versions at https://hub.docker.com/r/oven/bun/tags
+FROM oven/bun:1 AS base
+WORKDIR /app
+
+
+# -----------------------------------------------------------
+# install dependencies for dev and prod into temp directories
+# this will cache them and speed up future builds
```

Comment:
Just curious, with gh actions caching, [they found that just installing from scratch every time was actually faster than caching](https://github.com/oven-sh/setup-bun/issues/14#issuecomment-1714116221). I'm guessing the cache restoring in Docker's case is much faster than in gh actions case, but just wanted to check.

Reply:
For what it's worth, this Dockerfile, including that comment, is mostly from the official Bun docs, https://bun.sh/guides/ecosystem/docker.

I did attempt to add caching via bun's `--cache-dir` argument and a build-time volume when I saw that they hadn't actually implemented it, despite that comment. (It's a common misconception that using multi-stage builds somehow caches them outside of Docker's normal layer caching mechanism.)

Anyway, after a little testing without any form of caching I found that you're right, it's just as fast without the caching as it is with. I still find that strange, but hey, if it works I guess I can't complain.

Reply:
You're right, I tested and caching isn't actually making anything faster. (I thought perhaps its incredible speed was because it was secretly caching things to `~/.bun/install/cache`, which FWIW `bun` does by default, but even with that removed it's still as fast.) I went ahead and removed it.

Reply:
Here's some related discussion/investigation I did a few months back. My best understanding atm is that Bun is just ridiculously fast at the dependency resolution part, and that the actual network download of packages is really not the bottleneck in package managers.